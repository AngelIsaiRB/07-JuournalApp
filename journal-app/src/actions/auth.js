import { types } from "../types/types"
import {firebase, googleAuthProvider} from "../firebase/firebaseConfig"

export const startLoginEmailPassword = ()=>{
    return (dispatch) =>{
        setTimeout(() =>{
            dispatch(login(1234,"isai"));
        },3500)
    }
}

export const startGoogleLogin=()=>{
    return (dispatch)=>{
        firebase.auth().signInWithPopup(googleAuthProvider)
        .then(({user}) =>{
            dispatch(login(user.uid, user.displayName));
        });
    }
}

export const login =(uid, displayName)=>(
     {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    })
