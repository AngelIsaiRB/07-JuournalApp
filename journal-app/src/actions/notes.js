import { db } from "../firebase/firebaseConfig";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";
import Swal from 'sweetalert2'
import { fileUpload } from "../helpers/fileUpload";
// react-journal

export const startNewNote = ()=>{
    return async(dispatch, getState) =>{
        const uid = getState().auth.uid;
        // const {uid} = getState().auth;
        const newNote={
            title:"",
            body:"",
            date: new Date().getTime(),            
        }
        
        const doc = await db.collection(`${uid}/journal/notes`).add(newNote);

        dispatch(activeNote(doc.id,newNote ));
        dispatch(addNewNote(doc.id, newNote));

    }
}

export const activeNote =(id, note)=>(
    {
        type: types.notesActive,
        payload:{
            id,
            ...note
        }
    }
);

export const addNewNote=(id, note)=>({
    type:types.notesAddNew,
    payload:{
        id,
        ...note
    }
})

export const startLoadingNotes =(uid)=>{
    return async(dispatch)=>{
        const notes = await loadNotes(uid); 
        dispatch(setNotes(notes));  
    }
}

export const setNotes = (notes) =>({
    type:types.notesLoad,
    payload:notes,
});


export const StartSaveNote =(note)=>{
    return async(dispatch, getState) =>{
        const {uid} = getState().auth;
        if(!note.url){
            delete note.url;
        }
        const noteFireStore = {...note};
        delete noteFireStore.id;

        try {
            await db.doc(`${uid}/journal/notes/${note.id}`).update(noteFireStore);
            dispatch(refreshNote(note.id, noteFireStore));
            Swal.fire('Save!','','success')            
        } catch (error) {
            Swal.fire('Error!',error.data,'error')            
        }
    }
}

export const refreshNote =(id,note)=>({
    type:types.notesUpdate,
    payload:{
        id,
        note:{
            id,
            ...note,
        }
    }
})


export const startUploading =  (file)=>{
    return async(dispatch, getState)=>{

        Swal.fire({
            title:"Uploading...",
            text:"please wait...",
            allowOutsideClick:false,
            onBeforeOpen:() => {
                Swal.showLoading();
            }
        });

        const {active:activeNote} = getState().notes;
        const fileUrl = await fileUpload(file);
        activeNote.url=fileUrl;

        dispatch(StartSaveNote(activeNote));
        Swal.close();


    }
}

export const startDeleting =(id)=>{
    return  async (dispatch, getState) =>{
        const uid = getState().auth.uid;
        await db.doc(`${uid}/journal/notes/${id}`).delete();

        dispatch(deleteNote(id));

    }   
}

export const deleteNote = (id) =>({
    type:types.notesDelete,
    payload: id,
});


export const noteLogout =() =>({
    type:types.notesLogoutCleaning
})