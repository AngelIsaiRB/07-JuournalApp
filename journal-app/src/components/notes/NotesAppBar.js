import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { StartSaveNote, startUploading } from '../../actions/notes'
import moment from 'moment'

export const NotesAppBar = () => {

    const dispatch = useDispatch();
    const {active} = useSelector(state => state.notes)
    
    // string date
    const noteDate = moment(active.date);
    const dateString = `${noteDate.format("dddd")}-${noteDate.format("Do")} `;
    // 
    const handleSave =()=>{
        dispatch(StartSaveNote(active));
    }
    const handlePictureClick =()=>{
        document.querySelector("#fileSelector").click();
    }

    const handlefileChange=(e)=>{
        const file = e.target.files[0];
        if(file){
            dispatch(startUploading(file));
        }
    }

    return (
        <div className="notes__appbar"> 
            <span>{dateString}</span>
            <input type="file"
                id="fileSelector"
                name="file"
                style={{display:"none"}}
                onChange={handlefileChange}
            ></input>
            <div>
                <button 
                className="btn"
                onClick={handlePictureClick}
                >
                    picture
                </button>
                <button 
                    onClick={handleSave}
                    className="btn">
                    Save
                </button>
            </div>
        </div>
    )
}
