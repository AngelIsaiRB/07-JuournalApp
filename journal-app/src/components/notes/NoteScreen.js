import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    return (
        <div className="notes__main-content"> 
            <NotesAppBar/>

            <div className="notes__content">
                <input
                    className="notes__title-input"
                    type="text"
                    placeholder="some awesome title"
                    autocomplete="off"
                ></input>
                <textarea
                    className="notes__textarea"
                    placeholder="what happened today?"
                ></textarea>
                <div className="notes__image">
                    <img
                        src="https://miro.medium.com/max/3200/1*qyhua3giJFAx_kOdFoWHzw.jpeg"
                        alt="imagen"
                    >
                    </img>

                </div>


            </div>


        </div>
    )
}
