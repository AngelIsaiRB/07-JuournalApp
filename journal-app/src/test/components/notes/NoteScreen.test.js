import React from 'react';
import {  mount } from 'enzyme'

import configureStore from 'redux-mock-store'; 
import thunk from "redux-thunk";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { NoteScreen } from '../../../components/notes/NoteScreen';

import { activeNote } from '../../../actions/notes';
jest.mock('../../../actions/notes',()=>({
    activeNote : jest.fn(),
    
}))


const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
    auth:{
        uid:"12345",
        name:"isai"
    },
    ui:{
        loading: false,
        msgError: null,
    },
    notes:{
        active: {
            id:"1234",
            title:"hola",
            body:"mundo",
            date:0
        },
        notes:[]
    }
    
}

let store = mockStore(initState);
store.dispatch= jest.fn();
const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <NoteScreen/>
        </MemoryRouter>
    </Provider>
    );

describe('pruebas en <NotesScreen/>', () => {

    test('should  de mostrarse correctamete', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should de llamar el activeNote', () => {
        wrapper.find("input[name='title']").simulate("change",{
            target:{
                name:"title",
                value:"hola de nuevo"
            }
        });
        expect(activeNote).toHaveBeenLastCalledWith("1234",
        {
            body:"mundo",
            title:"hola de nuevo",
            id:"1234",
            date:0
        });
    })
    

})