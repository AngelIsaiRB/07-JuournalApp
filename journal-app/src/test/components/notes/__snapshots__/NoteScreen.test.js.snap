// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <NotesScreen/> should  de mostrarse correctamete 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      Wednesday-31st 
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="some awesome title"
      type="text"
      value="hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="what happened today?"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete!
  </button>
</div>
`;
